boilerplate/openshift/golang-osd-operator/csv-generate/csv-generate.sh

```diff
@@ -0,0 +1,127 @@
+            else
+                # TODO : Case to be tested
+                echo "Incorrect value for '-g'. Expecting 'hack' or 'common'. Got ${OPTARG}"
+            fi
+            ;;
+        H)
+            operator_commit_hash=${OPTARG}
+            ;;
+        i)
+            operator_image=${OPTARG}
+            ;;
+        n)
+            operator_commit_number=${OPTARG}
+            ;;
+        o)
+            operator_name=${OPTARG}
+            ;;
+        *)
+            usage
+    esac
+done
+
+# Checking parameters
+check_mandatory_params operator_channel operator_image operator_name operator_commit_hash operator_commit_number generate_script
+
+# If no override, using the gitlab repo
+if [ -z "$GIT_PATH" ] ; then 
+    GIT_PATH="https://app:@gitlab.cee.redhat.com/service/saas-${operator_name}-bundle.git"
+fi
+
+# Calculate previous version
+SAAS_OPERATOR_DIR="saas-${operator_name}-bundle"
+BUNDLE_DIR="$SAAS_OPERATOR_DIR/${operator_name}/"
+
+if [ "$diff_generate" = true ] ; then
+    OPERATOR_NEW_VERSION=$(ls "$BUNDLE_DIR" | sort -t . -k 3 -g | tail -n 1)
+    OPERATOR_PREV_VERSION=$(ls "${BUNDLE_DIR}" | sort -t . -k 3 -g | tail -n 2 | head -n 1)
+    OUTPUT_DIR="output-comparison"
+    
+    # For diff usecase, checking there is already a generated CSV
+    if [ ! -f ${BUNDLE_DIR}/${OPERATOR_NEW_VERSION}/*.clusterserviceversion.yaml ] ; then
+        echo "You need to generate CSV with your legacy script before trying to run the diff option"
+        exit 1 
+    fi
+else
+    rm -rf "$SAAS_OPERATOR_DIR"
+    git clone --branch "$operator_channel" ${GIT_PATH} "$SAAS_OPERATOR_DIR"
+    
+    # remove any versions more recent than deployed hash
+    if [[ "$operator_channel" == "production" ]]; then
+        if [ -z "$DEPLOYED_HASH" ] ; then
+            DEPLOYED_HASH=$(
+                curl -s "https://gitlab.cee.redhat.com/service/app-interface/raw/master/data/services/osd-operators/cicd/saas/saas-${_OPERATOR_NAME}.yaml" | \
+                    docker run --rm -i quay.io/app-sre/yq yq r - "resourceTemplates[*].targets(namespace.\$ref==/services/osd-operators/namespaces/hivep01ue1/${_OPERATOR_NAME}.yml).ref"
+            )
+        fi
+    
+        delete=false
+        # Sort based on commit number
+        for version in $(ls $BUNDLE_DIR | sort -t . -k 3 -g); do
+            # skip if not directory
+            [ -d "$BUNDLE_DIR/$version" ] || continue
+    
+            if [[ "$delete" == false ]]; then
+                short_hash=$(echo "$version" | cut -d- -f2)
+    
+                if [[ "$DEPLOYED_HASH" == "${short_hash}"* ]]; then
+                    delete=true
+                fi
+            else
+                rm -rf "${BUNDLE_DIR:?BUNDLE_DIR var not set}/$version"
+            fi
+        done
+    fi
+    # TODO : Clean handling of major version (should be variable from consumer repo and default to 0.1 is undefined)
+    OPERATOR_PREV_VERSION=$(ls "$BUNDLE_DIR" | sort -t . -k 3 -g | tail -n 1)
```

Comment:
Again here we should account for the case where we're building the first version.